### ISSUE TITLE
Doesn't appear to load main-bower-files

### ISSUE THREAD

**Issue by gin93r** [jackfranklin/gulp-load-plugins#120]
```
// include gulp
var gulp       = require("gulp");

// include plugins
var plugins = require("gulp-load-plugins")({
    patterns : ["gulp-*", "gulp.*", "main-bower-files"],
    replaceString: /\bgulp[\-.]/
});
```

If I log plugins, mainBowerFiles is not listed and calling `plugins.mainBowerFiles()` errors with "mainBowerFiles" is not a function.

Thoughts?

**Comment by jackfranklin** [comment]
Can you get the project up online?

And do you have `main-bower-files` in your `package.json` as a dependency?

You could pass: `DEBUG: true` as an option and look at what gets outputted.

**Comment by gin93r** [comment]
I'll see if I can get it uploaded. It's saved as a devDependency though. I'll pass DEBUG : true when I get in.

**Comment by jackfranklin** [comment]
Should be fine as a dev dependency.

**Comment by gin93r** [comment]
Ok, I got the [project online](https://github.com/gin93r/baseproject).

I passed `DEBUG : true` and it looks like it's loading the plugin (9 in total), though it's only lazy loading 6 of them. Here's a screenshot of the output from the debug.

![image](https://cloud.githubusercontent.com/assets/5880428/18128735/4aa88918-6f55-11e6-878c-d1a7f641b58e.png)


Let me know if there's anything else you need.

**Comment by jackfranklin** [comment]
It looks like it's not renaming... is `plugins['main-bower-files']` defined?

**Comment by gin93r** [comment]
Negative. Using `plugins['main-bower-files']` produces the same error, just with `main-bower-files` instead of `mainBowerFiles`

**Comment by jackfranklin** [comment]
Can you disable lazy loading and see if that fixes?

**Comment by gin93r** [comment]
Same result with both `plugins.mainBowerFiles` and `plugins['main-bower-files']` with `lazy : false`

**Comment by jackfranklin** [comment]
OK. I don't really have time to debug this right now. I'd suggest you edit `node_modules/gulp-load-plugins/index.js`, add some `console.log` calls and the like and try to figure it, and let me know if you get anywhere. Else I'll try to look when I next have time.

**Comment by gin93r** [comment]
No worries. Thanks for your time.

**Comment by jackfranklin** [comment]
If you do find anything of note do let me know :)

**Comment by LTheobald** [comment]
Hi @gin93r. It looks like you made the same mistake as I did. Line 6 in your example:

    patterns : ["gulp-*", "gulp.*", "main-bower-files"],

The key needs to be `pattern` not `patterns`. It's failing back to the default pattern which is the `gulp-` prefixed items & hence why you don't see `mainBowerFiles()`

**Comment by jackfranklin** [comment]
@LTheobald awesome, thank you! I'm sure we could put some nicer logging in if you supply an unknown option. I'll open an issue for that.

**Comment by gin93r** [comment]
@LTheobald Oh - wow. Thanks! Such a simple oversight. I'm glad it was a user error though.

**Comment by jackfranklin** [comment]
Closing this but tracking a potential fix in #123, which anyone is welcome to take as a PR :)
